src/tools/Avalonia.Designer.HostApp/Program.cs

```diff
@@ -1,10 +1,43 @@
-﻿
+﻿using System;
+using System.IO;
+using System.Reflection;
 
 namespace Avalonia.Designer.HostApp
 {
     class Program
     {
-        public static void Main(string[] args) 
-            => Avalonia.DesignerSupport.Remote.RemoteDesignerEntryPoint.Main(args);
+        private static string s_appDir;
```

Comment:
Are you sure that it works properly on .NET Core? I don't quite remember the reason why I did it that way, but I think that on .NET Core dependencies have to be resolved from deps.json rather than some arbitrary directory.

Reply:
It works. It’s actually required on .NET Core with this PR because of the refactoring of the DotNetCoreRuntime assembly to not be crash prone.

The backends are now not directly referenced, so they aren’t in the deps.json file. And the assembly search in UsePlatformDetect uses the app entry point directory, which is correct in all cases except this case. So we install this handler to enable the assembly resolution to work.

Reply:
>The backends are now not directly referenced, so they aren’t in the deps.json file.

That's probably a bad idea, since most of the tooling kinda expects *everything* required to be present in .deps.json.

Reply:
BTW, how does it even work? Are platform backends get copied to the application output dir? Usually `bin/$(Configuration)/netcoreapp*.*` doesn't contain anything from nuget packages, only app itself and its project dependencies.

Reply:
It was testing via project references. I haven't run the test with a package reference I realize. You might be right on us having an issue.

Reply:
Maybe I could move `UsePlatformDetect` into `Avalonia.Desktop` and use the old implementation that has direct assembly references.

Reply:
Avalonia.Desktop seems to be a proper place for UsePlatformDetect